core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -1376,8 +1376,14 @@ class CivilizationInfo : IsPartOfGameInfoSerialization {
     }
 
     fun moveCapitalToNextLargest() {
-        moveCapitalTo(cities
-            .filterNot { it.isCapital() }
+        val availableCities = cities.filterNot { it.isCapital() }
+        if (availableCities.filterNot { it.isPuppet }.isEmpty()) {
+            availableCities
+                .filter { it.isPuppet }
+                .maxByOrNull { it.population.population }!!.annexCity()
```

Comment:
See Yair's comment, although I think it would be possible to change this line to `.maxByOrNull { it.population.population }?.annexCity()` and have the code still work but I haven't checked

Reply:
I used null assertion since I think it should crash if it entered the if statement but didn't manage to annex a city